kafka-spring-boot-autoconfigure/src/main/java/org/axonframework/extensions/kafka/autoconfig/KafkaAutoConfiguration.java

```diff
@@ -148,4 +152,41 @@ public Fetcher kafkaFetcher(ConsumerFactory<String, byte[]> kafkaConsumerFactory
     public KafkaMessageSource kafkaMessageSource(Fetcher kafkaFetcher) {
         return new KafkaMessageSource(kafkaFetcher);
     }
+
+    @ConditionalOnBean({ EventProcessingConfigurer.class })
```

Comment:
I believe the `@Conditional` annotation only takes effect on type-level, as a meta-annotation and on methods annotated with `@Bean`.